pkg/users/local.js

```diff
@@ -47,6 +47,24 @@
     });
 };
 
+export function get_next_available_id(already_used, is_groups, change) {
+    //  Handle also the case where logindef == null, i.e. the file does not exist.
+    //  While that's unusual, "empty /etc" is a goal, and it shouldn't crash the page.
+    cockpit.file("/etc/login.defs").read()
+            .then(logindef => {
+                const ids = already_used.filter(a => a.name !== 'nobody').map(a => is_groups ? a.gid : a.uid);
+
+                const min_regex = is_groups ? /^GID_MIN\s+(\d+)/m : /^UID_MIN\s+(\d+)/m;
+                const max_regex = is_groups ? /^GID_MAX\s+(\d+)/m : /^UID_MAX\s+(\d+)/m;
+                const min = parseInt((logindef && logindef.match(min_regex)[1]) || 500);
+                const max = parseInt((logindef && logindef.match(max_regex)[1]) || 60000);
+
+                change("min_id", min);
+                change("max_id", max);
+                change(is_groups ? "id" : "uid", Math.max(min, Math.max(...ids.filter(id => id < max)) + 1).toString());
+            }, () => Promise.resolve());
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18353-20230222-095918-e839713c-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)